src/compiler/tracing.ts

```diff
@@ -88,38 +93,39 @@ namespace ts.tracing {
         Emit = "emit",
     }
 
-    export function begin(phase: Phase, name: string, args: object) {
-        if (!traceFd) {
-            return;
-        }
-        Debug.assert(fs);
+    export function begin(phase: Phase, name: string, args?: object) {

```

Comment:
`/** Note: `push` should be used by default.  `begin` is for special cases where we need the data point even if the event never  terminates (typically for reducing a scenario too big to trace to one that can be completed). */`

Reply:
Done, tweaked a bit.